src/main/java/appeng/container/implementations/ContainerInterfaceTerminal.java

```diff
@@ -92,22 +88,20 @@ public void detectAndSendChanges() {
         if (host != null) {
             final IGridNode agn = host.getActionableNode();
             if (agn != null && agn.isActive()) {
-                for (final IGridNode gn : this.grid.getMachines(TileInterface.class)) {
-                    InterfaceCheck interfaceCheck = new InterfaceCheck().invoke(gn);
-                    total += interfaceCheck.getTotal();
-                    missing |= interfaceCheck.isMissing();
-                }
-
-                for (final IGridNode gn : this.grid.getMachines(PartInterface.class)) {
-                    InterfaceCheck interfaceCheck = new InterfaceCheck().invoke(gn);
-                    total += interfaceCheck.getTotal();
-                    missing |= interfaceCheck.isMissing();
-                }
-
-                for (final IGridNode gn : this.grid.getMachines(PartP2PInterface.class)) {
-                    InterfaceCheck interfaceCheck = new InterfaceCheck().invoke(gn);
-                    total += interfaceCheck.getTotal();
-                    missing |= interfaceCheck.isMissing();
+                var clzWithInterfaceSupport = StreamSupport.stream(grid.getMachinesClasses().spliterator(), false)
+                        .filter(IInterfaceTerminalSupport.class::isAssignableFrom).distinct().toArray(Class[]::new);
```

Comment:
I do not like how it needs to constantly check every single machine type on the Grid every time `detectAndSendChanges` is called and create a new class list every time. Having mods register their classes at init/postInit is better.

On the other hand, a major refactor needs to happen here anyways given the performance complaints that we've received already.

Reply:
I think that shouldn't be too bad performance-wise? There are not too many machine types and it probably won't be the slowest thing in the code right?

Reply:
We'll need to rework this after stable

Reply:
done